### PR TITLE
Add libsecret-based pfPasswordStore backend for non-OSX unixes

### PR DESCRIPTION
**Pull request by zrax**
It compiles, but is otherwise untested since we don't actually have a *nix plClient yet...  But this should be a step in that direction.

### CODE REVIEW

**Comment by dpogue** [comment]
😍

File: Sources/Plasma/FeatureLib/pfPasswordStore/CMakeLists.txt [hunk 1]
```diff
@@ -38,6 +39,10 @@ if(APPLE)
     target_link_libraries(pfPasswordStore ${SECURITY_LIBRARY})
 endif(APPLE)
 
+if(UNIX AND NOT APPLE)
+    target_link_libraries(pfPasswordStore ${LIBECRET_LIBRARIES})
```
Comment (dpogue) [inline Sources/Plasma/FeatureLib/pfPasswordStore/CMakeLists.txt]: typo: LIB**S**ECRET

File: CMakeLists.txt [hunk 2]
```diff
@@ -175,6 +175,11 @@ if(WIN32)
     )
 endif(WIN32)
 
+if(UNIX AND NOT APPLE)
+    find_package(PkgConfig REQUIRED)
+    pkg_check_modules(LIBSECRET REQUIRED libsecret-1)
```
Comment (dpogue) [inline CMakeLists.txt]: Can we make this optional and fallback to file-based storage if libsecret isn't installed? I suspect it'll be available on almost every modern *nix, but it's nice to not have additional hard dependencies

File: Sources/Plasma/FeatureLib/pfPasswordStore/pfPasswordStore_impl.h [hunk 3]
```diff
@@ -75,10 +75,19 @@ class pfWin32PasswordStore : public pfPasswordStore
...
 #endif //HS_BUILD_FOR_WIN32
 
-
+#if defined(HS_BUILD_FOR_UNIX) && !defined(HS_BUILD_FOR_OSX)
 /**
-* @todo A Linux libsecret-based storage mechanism.
-*/
+ * A libsecret-based password storage mechanism.
+ */
+class pfUnixPasswordStore : public pfPasswordStore
```
Comment (dpogue) [inline Sources/Plasma/FeatureLib/pfPasswordStore/pfPasswordStore_impl.h]: You need to update pfPasswordStore.cpp to actually create a pfUnixPasswordStore when we ask for an instance

File: Sources/Plasma/FeatureLib/pfPasswordStore/pfPasswordStore_Unix.cpp [hunk 4]
```diff
@@ -43,16 +43,58 @@ You can contact Cyan Worlds, Inc. by email legal@cyan.com
...
+    return result;
 }
 
-bool pfUnixPasswordStore::SetPassword(const ST::string& username, const ST::string& password)
+bool pfUnixPasswordStore::SetPassword(const ST::string& username,
+                                      const ST::string& password)
 {
+    GError *error = nullptr;
+    secret_password_store_sync(&pfPasswordStore_Schema, SECRET_COLLECTION_DEFAULT,
+                               "Myst Online Password", password.c_str(),
```
Comment (dpogue) [inline Sources/Plasma/FeatureLib/pfPasswordStore/pfPasswordStore_Unix.cpp]: Should the "Myst Online Password" include the server and/or username in the description?

The general pattern I see in Seahorse for Gnome apps is "[program name] password for [account ID]"

**Comment by zrax** [comment]
Updated to address review comments

**Pull request merged**